docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -335,7 +335,18 @@ void TemplateInferRequest::inferPostprocess() {
         }
         auto outputBlob = _outputs.at(output.first);
         auto networkOutput = _networkOutputBlobs[output.first];
-        if (outputBlob->getTensorDesc().getPrecision() != networkOutput->getTensorDesc().getPrecision()) {
+        if (result->get_output_partial_shape(0).is_dynamic() &&
```

Comment:
Why do we need this branch?

Reply:
There are two branches, the first exists in case precision networkOutput and outputBlob do not coincide, and the second exists in case shapes of results are dynamic. The case of simultaneous existence of these two events was not handled.